### ISSUE TITLE
Error caused by numeric ID's

### ISSUE THREAD

**Issue by aquavark** [select2/select2#5810]
Hi

I believe this a bug, as numeric only ID's is acceptable per standards.

If you use numeric ID's for a select (that work in normal selects and in select2 up to 4.0.5) in 4.0.6 to 4.0.13, the the select box becomes corrupted and won't select properly.

Minimal example I've been able to build:

```
<html>
<head>
    <script src="shared/jquery/jquery-3.4.1.js"></script>
    <link rel='stylesheet' type='text/css' href='shared/jquery/select2-4.0.12/css/select2.css'/>
    <script src="shared/jquery/select2-4.0.12/js/select2.full.js"></script>
    <script>
    $(function() {     
        $('#shiftsel').select2();
    });
</script>
</head>
<body>
<select id="shiftsel" tabindex="-1" ><option id='4' value="3_DE8_A_1_1">DE8 - Day 0700 8 (07:00-15:30)</option><option id='0' value="18_DE9_~_9_1">DE9 - Day 0700 9 (07:00-16:30)</option><option id='5' value="4_DG8_B_1_1">DG8 - Day 0730 8 (07:30-16:00)</option></select>
</body>
</html>
```
In this example, tested in Firefox / Linux, when you select items from the select box the text gets changed to "Searching..." (in pale gray), and subsequent selects act strangely.

This was first seen following an upgrade in a larger (private) site to Select2 4.0.12, where a page with multiple select2's started acting strangely - items went missing, and items from one select started appearing in the other select.

If all of the id='<number only>' attributes are removed Select2 functions normally again.

Happy to provide further information

Richard

**Comment by aquavark** [comment]
Additional - once it errors, you start seeing errors in the console on further selects:

14:49:15.162 TypeError: s.id is undefined select2.full.js:965:9
    jQuery 21
        selectedIds
        map
        setClasses
        current
        setClasses
        bind
        invoke
        trigger
        trigger
        _registerEvents
        invoke
        trigger
        trigger
        open
        toggleDropdown
        _registerSelectionEvents
        invoke
        trigger
        bind
        dispatch
        handle

**Comment by kevin-brown** [comment]
Any chance you can bundle this into a jsbin or jsfiddle just so we can get a live example to test with?

I appreciate it being traced down to 4.0.6 though, because that gives us a solid point to start from. Do you know if it works/fails in any of the 2 release candidates for 4.0.6 (-rc.0 or -rc.1)? That was one of our larger releases.

**Comment by kevin-brown** [comment]
It just occurred to me that this probably has to do with Select2's internal ID generation. Do you happen to have duplicate IDs on the page anywhere?

**Comment by aquavark** [comment]
The snippet above is complete and shows a fault in itself. In the private site, there are no duplicate ID's that I can see on the page. It has 3 Select2's, however, and in that case the corruption occurs with data across the three selects (so items from one select dataset appear in the other, I've not discerned the pattern). The one that had numeric ID's appears to be the trigger, but it corrupts the other one as well. They are dynamically loaded, however in the example given it's direct. The other two lists didn't have ID's at all.

And yes, 4.0.6-RC1 has the bug (I happened to find this on the system), and it's present in 4.0.12 and 4.0.13 for sure. Although not an expert on your code, I did see that 4.0.6 apparently introduced a different way of storing the item data.

Hopefully, this works on JSBin - it definitely shows a fault once you've selected your first item. I also noted that some values for ID are worse than others - setting one to ID 4 causes issues when other numerics don't, but this may be particular to the datasets. Adding more to the select doesn't change that ID 4 breaks it, but which one is ID'd as 4 in this case changes where it breaks.

https://jsbin.com/wemaposejo/edit?html,output

Otherwise, this is the JSBin code

`<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width">
  <title>JS Bin</title>
</head>
<body>
<script src="https://code.jquery.com/jquery-3.1.0.js"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/select2/4.0.12/js/select2.full.min.js"></script>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/select2/4.0.12/css/select2.min.css"  type='text/css'/>
<script> $(function() {  $('#shiftsel').select2();  }); </script>
  
  <select id="shiftsel" tabindex="-1" ><option id='4' value="3_DE8_A_1_1">DE8 - Day 0700 8 (07:00-15:30)</option><option id='0' value="18_DE9_~_9_1">DE9 - Day 0700 9 (07:00-16:30)</option><option id='5' value="4_DG8_B_1_1">DG8 - Day 0730 8 (07:30-16:00)</option></select>
</body>
</html>`

Many thanks for looking at this

**Comment by kevin-brown** [comment]
> Although not an expert on your code, I did see that 4.0.6 apparently introduced a different way of storing the item data.

Yeah, this is exactly where the issue is occurring. Select2 is generating an internal ID that is conflicting with the existing IDs on the page. I'm working on a fix that will go out in the upcoming 4.1.0 release.
